### DISCUSSION TITLE
An idea for an mcu.  STM32L1

### DISCUSSION THREAD

**Discussion by BOBBYWY** [libresmart/denada#2] [Ideas]
https://www.st.com/content/st_com/en/products/microcontrollers-microprocessors/stm32-32-bit-arm-cortex-mcus/stm32-ultra-low-power-mcus/stm32l1-series.html

    Ultra-low-power mode: 280 nA with backup registers (3 wakeup pins)
    Ultra-low-power mode + RTC: 900 nA with backup registers (3 wakeup pins)
    Low-power run mode: down to 9 μA
    Dynamic run mode: down to 177 μA/MHz

**Comment by libresmartowner** [comment]
Can we make the code generic enough that we could switch out the specific microcontroller and only do a small amount of configuration changes to compile for different hardware?

I ask because I initially envisioned this running on a SparkFun Artemis.

**Comment by tekemperor** [comment]
> https://www.st.com/content/st_com/en/products/microcontrollers-microprocessors/stm32-32-bit-arm-cortex-mcus/stm32-ultra-low-power-mcus/stm32l1-series.html
> 
> ```
> Ultra-low-power mode: 280 nA with backup registers (3 wakeup pins)
> Ultra-low-power mode + RTC: 900 nA with backup registers (3 wakeup pins)
> Low-power run mode: down to 9 μA
> Dynamic run mode: down to 177 μA/MHz
> ```

I had an ESP32 leftover from another project and created a pull request with some of the basics that were missing (like LCD support, and a keyboard input buffer). The USB host library I used for keyboard support is unique to that MCU. Everything else should be portable. However, I don't currently have anything else to develop or test with.

I am fairly new to hardware development, any feedback or assistance would be appreciated. Is there any particular development board/kit you recommend for the STM32L1?

PR: https://github.com/libresmart/denada/pull/4

**Reply by BOBBYWY** [reply]
Sorry, nothing to recommend on the STM32L.  Came across it while looking for some other bits.  Playing with some oddball Intel stuff now, not something I would recommend for this.
